internal/transport/http/http.go

```diff
@@ -19,77 +20,80 @@ type Transport struct {
 	controlHandler  transport.CommandHandler
 	dataHandler     transport.DataHandler
 	pollingInterval time.Duration
-	killSwitch      chan struct{}
+	disconnected    atomic.Value
 }
 
 func NewHTTPTransport(ClientID string, server string, tlsConfig *tls.Config, userAgent string,
 	pollingInterval time.Duration, controlHandler transport.CommandHandler,
 	dataHandler transport.DataHandler) (*Transport, error) {
+	disconnected := atomic.Value{}
+	disconnected.Store(false)
 	return &Transport{
 		Server:          server,
 		ClientID:        ClientID,
 		HttpClient:      http.NewHTTPClient(tlsConfig, userAgent),
 		controlHandler:  controlHandler,
 		dataHandler:     dataHandler,
 		pollingInterval: pollingInterval,
+		disconnected:    disconnected,
 	}, nil
 }
 
-func (t Transport) Start() error {
-	t.killSwitch = make(chan struct{})
+func (t *Transport) Start() error {
+	t.disconnected.Store(false)
 	go func() {
 		for {
-			select {
-			case <-t.killSwitch:
+			if t.disconnected.Load().(bool) {
 				return
-			default:
-				payload, err := t.HttpClient.Get(t.getUrl("in", "control"))
-				if err != nil {
-					log.Errorf("Error while getting work: %v", err)
-				}
-				if payload != nil && len(payload) > 0 {
-					t.controlHandler(payload, t)
-				}
-				time.Sleep(t.pollingInterval)
 			}
+			payload, err := t.HttpClient.Get(t.getUrl("in", "control"))
+			if err != nil {
+				log.Tracef("Error while getting work: %v", err)
+			}
+			if payload != nil && len(payload) > 0 {
+				t.controlHandler(payload, t)
+			}
+			time.Sleep(t.pollingInterval)
 		}
 	}()
 
 	go func() {
 		for {
-			select {
-			case <-t.killSwitch:
+			if t.disconnected.Load().(bool) {
 				return
-			default:
-				payload, err := t.HttpClient.Get(t.getUrl("in", "data"))
-				if err != nil {
-					log.Errorf("Error while getting work: %v", err)
-				}
-				if payload != nil && len(payload) > 0 {
-					t.dataHandler(payload)
-				}
-				time.Sleep(t.pollingInterval)
 			}
+			payload, err := t.HttpClient.Get(t.getUrl("in", "data"))
+			if err != nil {
+				log.Tracef("Error while getting work: %v", err)
+			}
+			if payload != nil && len(payload) > 0 {
+				t.dataHandler(payload)
+			}
+			time.Sleep(t.pollingInterval)
 		}
 	}()
 
 	return nil
 }
 
-func (t Transport) SendData(data yggdrasil.Data) error {
+func (t *Transport) SendData(data yggdrasil.Data) error {
 	return t.send(data, "data")
 }
 
-func (t Transport) SendControl(ctrlMsg interface{}) error {
+func (t *Transport) SendControl(ctrlMsg interface{}) error {
 	return t.send(ctrlMsg, "control")
 }
 
-func (t Transport) Disconnect(quiesce uint) {
+func (t *Transport) Disconnect(quiesce uint) {
 	time.Sleep(time.Millisecond * time.Duration(quiesce))
-	t.killSwitch <- struct{}{}
+	t.disconnected.Store(true)
 }
 
-func (t Transport) send(message interface{}, channel string) error {
+func (t *Transport) send(message interface{}, channel string) error {
+	disconnected := t.disconnected.Load()
+	if disconnected.(bool) {
```

Comment:
I see in other places you use `if t.disconnected.Load().(bool) {`. Do you think it make sense to have it similar here or is there any benefit of having it in 2 lines?

Reply:
You're right; there is no reason to have it  in a separate line.

Reply:
Done